sentry_sdk/profiler.py

```diff
@@ -137,22 +135,52 @@ def _extract_stack(frame):
     only the first `MAX_STACK_DEPTH` frames will be returned.
     """
 
-    stack = deque(maxlen=MAX_STACK_DEPTH)  # type: Deque[FrameData]
+    stack = deque(maxlen=MAX_STACK_DEPTH)  # type: Deque[FrameType]
 
     while frame is not None:
-        stack.append(
-            (
-                # co_name only contains the frame name.
-                # If the frame was a class method,
-                # the class name will NOT be included.
-                frame.f_code.co_name,
-                frame.f_code.co_filename,
-                frame.f_code.co_firstlineno,
-            )
-        )
+        stack.append(frame)
         frame = frame.f_back
 
-    return stack
+    return [
+        FrameData(
+            name=get_frame_name(frame),
+            file=frame.f_code.co_filename,
+            line=frame.f_lineno,
+        )
+        for frame in stack
+    ]
```

Comment:
Instead of extracting the data for every frame while we step through the stack, we step through the stack first and only extract the data from the frames within the `MAX_STACK_DEPTH`.

Some quick benchmarks on 100,000 iterations
- stack depth of 100: 4.22s vs 4.55s (a little slower)
- stack depth of 500: 21.76s vs 8.32 (a lot faster)

Since the slowest part of this is the `get_frame_name` function and I figured it was worth the trade off here to optimize for the worst case and be a little slower on the common cases.